merlin-sdk/src/main/java/gov/nasa/jpl/aerie/merlin/protocol/types/DurationType.java

```diff
@@ -16,4 +19,12 @@ static DurationType controllable(final String parameterName) {
   static DurationType fixed(final Duration duration) {
     return new Fixed(duration);
   }
+
+  static DurationType parametric(final ThrowingDurationFunction durationFunction) {
+    return new Parametric(durationFunction);
+  }
+
+  interface ThrowingDurationFunction {
```

Comment:
Could there be a non-throwing `DurationFunction`? If not, I think `Throwing` should be removed from the interface name.

Reply:
No there can't. I chose that because in java the traditional functional interfaces don't accept functions that throw checked exceptions. I wanted to just use `Function<Map<String, SerializedValue>, Duration>`. I called it `Throwing` because the fact that it throws is the reason its needed.